src/formio/components/file.ts

```diff
@@ -79,7 +79,7 @@ export interface FileUploadConfiguration {
 export interface BaseFileComponentSchema
   extends Omit<StrictComponentSchema<FileUploadData[]>, UnusedFileProperties | 'errors'>,
     DisplayConfig,
-    Omit<OFExtensions<TranslatableKeys>, 'registration'>,
+    Omit<OFExtensions<TranslatableKeys, {softRequired?: boolean}>, 'registration'>,
```

Comment:
Why add the `softRequired` to the openForms key? Why not, along with `required`, in the validation?

Reply:
at first I leaned towards that too, but it doesn't make sense because it's not a validator in Formio terms - we don't actually have a validation function running anywhere, and a validation error (produced by the validator function) causes the form to be in the `invalid` state, which blocks is from being submittable.

Additionally, through the types and existing functionality, this implies that you must be able to specify a (translatable) error message for that error/validator key, but we don't actually plan to display validation errors with the field itself (or if we do, semantically we should use the `required` message), but instead the errors will be collected in the new component typically displayed at the end of the step above the submit button.

So, at that point I'd be fighting the type system and fighting formio (runtime behaviour) to make it work by putting it in the `validate` object. Since this is an entirely custom concept foreign to Formio itself, it makes a lot more sense to drop it in our own namespace and implement the custom behaviour accordingly.

Reply:
Mhh okay, yeah in that case it may indeed be better to leave the `softRequired` as a completely custom part.. If adding it to `validate` cases so much trouble, it isn't worth it

And, indeed, this way it's obviously openForms custom logic